src/common/datatypes/Geography.h

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include <s2/s2point.h>
+#include <s2/s2point_region.h>
+#include <s2/s2polyline.h>
+#include <s2/s2region.h>
+
+#include <algorithm>
+#include <typeinfo>
+#include <vector>
+
+#include "common/base/StatusOr.h"
+#include "common/datatypes/Value.h"
+#include "common/geo/io/Geometry.h"
+
+// Do not include <s2/s2polygon.h> here, it will indirectly includes a header file which defines a
+// enum `BEGIN`(not enum class). While Geography.h is indirectly included by parser.yy, which has a
+// macro named `BEGIN`. So they will be conflicted.
+
+class S2Polygon;
+
+namespace nebula {
+
+// clang-format off
+/*
+static const std::unordered_map<GeoShape, S2Region> kShapeTypeToS2Region = {
+    // S2PointRegion is a wrapper of S2Point, and it inherits from the S2Region class while S2Point doesn't.
+    {GeoShape::POINT, S2PointRegion},
+    {GeoShape::LINESTRING, S2Polyline},
+    {GeoShape::POLYGON, S2Polygon},
+};
+*/
+// clang-format on
```

Comment:
cleanup if useless

Reply:
It's not useless, It's note, which is just to illustrate the mapping between GeoShape and S2 shape